sesame/middleware.py

```diff
@@ -11,15 +15,23 @@ class AuthenticationMiddleware:
 
     def __init__(self, get_response):
         self.get_response = get_response
+        self._redirect_url = None
 
     def __call__(self, request):
+        self._redirect_url = None
         self.process_request(request)
-        return self.get_response(request)
+        response = self.get_response(request)
+
+        if self._redirect_url:
```

Comment:
I'd prefer that `self.process_request` return `True` when a user was logged in and `False` otherwise. That would avoid relying on the side effect of setting a variable.